### DISCUSSION TITLE
Problem with refetchInterval

### DISCUSSION THREAD

**Discussion by aericson** [TanStack/query#399] [General] [answered]
Hello,

was wondering if someone could help me figure out what's going on. I've tried to reproduce it on the codesandbox bellow.

It's supposed to poll an API every 1s but it doesn't (see it on the console). I'm guessing it's because the `count` prop is being updated too often. If I stop changing the `count` in `App` it works fine.

I'm sure I'm just misunderstanding something here but would be nice to understand what's going on.

Thanks for the help!

codesandbox:
https://codesandbox.io/s/practical-wozniak-73cih

**Comment by tannerlinsley** [comment]
When the interval amount changes, it cancels the previously pending interval and starts a new one, so if your interval amount is changing faster than the interval time itself, it will never fire.

**Comment by tannerlinsley** [comment]
Beyond that, if you aren't changing the interval itself, then this could be a bug.

**Comment by aericson** [comment]
@tannerlinsley thanks! yea, the interval it's not changing just the component getting re-rendered due to another unrelated prop changing. If you get some time check the code-sandbox. Maybe it's a bug then.

**Comment by cherniavskii** [comment, marked as answer]
I checked the codesandbox example and confirm that something weird is going on, when parent rerenders often.
Will look into this later.

**Reply by cherniavskii** [reply]
Found the problem, working on a fix!

**Reply by aericson** [reply]
Big thanks!

**Reply by cherniavskii** [reply]
BTW @tannerlinsley , this discussion can be converted to the issue

**Reply by cherniavskii** [reply]
https://github.com/tannerlinsley/react-query/pull/404
